_posts/2022-06-01-snitching-on-your-colleagues-using-cloud-magic.md

```diff
@@ -0,0 +1,332 @@
+---
+layout: post
+authors: [lander_marien]
+title: 'Snitching On Your Colleagues Using Cloud Magic'
+image: /img/2022-06-01-snitching-on-your-colleagues-using-cloud-magic/Betrayal.jpg
+tags: [Spring, Spring Boot, Git, DevOps, Docker, Kubernetes, Terraform, Helm, Azure, Internet Of Things, RaspberryPi, Javascript, Node.js, AWS, DynamoDB, RDS, Postgres]
+category: Internship
+comments: true
+---
+
+
+
+# Table Of Contents
+
+* [Introduction](#introduction)
+  
+* [The Task At Hand](#the-task-at-hand)
+* [Exploring The Environment](#exploring-the-environment)
+  * [Azure Repos](#azure-repos)
+  * [Pipeline](#pipeline)
+  * [Library](#library)
+  * [Service Connections](#service-connections)
+* [Into The Rabbit-Hole : Setting Up The Pipeline](#into-the-rabbit-hole-setting-up-the-pipeline)
+  * [Before We Dive In](#before-we-dive-in)
+  * [Basic Tasks](#basic-tasks)
+    * [Maven Test](#maven-test)
+    * [Maven Build](#maven-build)
+    * [Copying And Creating An Artifact](#copying-and-creating-an-artifact)
+  * [Containerization](#containerization)
+    * [Fetching Our Artifact](#fetching-our-artifact)
+    * [Creating An Image](#creating-an-image)
+  * [Setting Up A Cloud Database](#setting-up-a-cloud-database)
+    * [Terraform Install](#installing-terraform)
+    * [Terraform Init](#initializing-terraform)
+    * [Terraform Plan](#terraform-plan)
+    * [Terraform Apply](#terraform-apply)
+  * [Provisioning A Cloud Cluster](#provisioning-a-cloud-cluster)
+    * [HELM](#helm-deploy)
+  * [Giving A Signal](#giving-a-signal)
+    * [Creating Our Bot](#creating-our-bot)
+    * [Sending Out Notifications](#sending-out-notifications)
+  * [A Visual Representation](#a-visual-representation)
+* [Out Of The Frying Pan And Into The Fire: Creating Our Own Task](#out-of-the-frying-pan-and-into-the-fire-creating-our-own-task)
+  * [Requirements](#requirements)
+  * [Getting The Logs](#getting-the-logs)
+  * [Filtering And Saving Failures](#filtering-and-saving-failures)
+  * [Factory Fresh](#factory-fresh)
+* [The Berry On Top: Our Physical Feedback](#the-berry-on-top--our-physical-feedback)
+  * [Dollar Store Google Assistant](#dollar-store-google-assistant)
+* [The Good, The Bad & ... The Ugly?](#the-good-the-bad---the-ugly--summary)
+  
+
+# Introduction
+
+7 weeks ago I started my internship at Ordina Mechelen.
+I had several project options available all looking to touch new and unknown tech that might be relevant for future operations.
+
+My inner Judas spoke to me when I saw a listing about a project that would shoot rockets at developers if they broke a build, and it would provide a great opportunity to pull myself out of my comfort zone focusing more on Devops and Cloud platforms rather than pure programming.
```

Comment:
```suggestion
My inner Judas spoke to me when I saw a listing about a project that would shoot toy rockets at developers if they broke a build, and it would provide a great opportunity to pull myself out of my comfort zone focusing more on Devops and Cloud platforms rather than pure programming.
```

Reply:
Especially with the current state of the world, we might want to be very clear here 😉